### ISSUE TITLE
Introduce an indices object

### ISSUE THREAD

**Issue by bdestombe** [dtscalibration/python-dts-calibration#122]
**Is your feature request related to a problem? Please describe.**
Something that crossed my mind several times, to have an object that returns parameter indices and measurement indices. The names of the indices of how it is currently implemented is vague to say the least (`ix_sec`, `ix_from_cal_match_to_glob`). 

It should clarify and clean up methods like `calibration_single_ended_solver()`, `calibration_double_ended_solver()`, `calibration_single_ended()`, and `calibration_double_ended()`.

**Describe the solution you'd like**
An object that is constructed with arguments like: `nx`, `nt`, `sections`, `matching_indices/matching_sections`, `transient_asym_att_x`.

**Comment by BSchilperoort** [comment]
Sounds like a good idea to clean up the code and make it more readable. Perhaps a method under the DataStore class?

eg.
```python
class DataStore(xr.Dataset):
    "Docs etc."

    def indices(self, index_name):
        if index_name == 'sections':
            return section_indices
```
As it's a method of `DataStore`, it should know all about the required arguments already, right?

It can then be used in the code;
```python
self.isel(x=self.indices('sections'))
```

**Comment by bdestombe** [comment]
That would definitely be a clean way to implement it.

Or register many properties. This way, all the properties enter nicely in the documentation.

**Comment by bdestombe** [comment]
A `DataStore` instance doesn't have all data btw. It still lacks `matching_indices/matching_sections` and `transient_asym_att_x`. But maybe we should treat those similar as `sections` and make them part of `DataStore`?

**Comment by BSchilperoort** [comment]
I think making the matching sections and transient x part of `DataStore` is a good idea. Do you foresee any compatibility issues in doing so?

I think that it should go OK if you can still pass a (different) `transient_asym_att_x` to the calibration function.
E.g. within the `calibration_double_ended` routine;
```python
if transient_asym_att_x is None:
    transient_asym_att_x = self.transient_asym_att_x
```
That would make everything backwards compatible. I think `sections` work in a similar way at the moment

**Comment by bdestombe** [comment]
Perfect. I'll come up with a PR later today.

The values would than be stored as `self.transient_asym_att_x` and as coordinates in `self.trans_att`. Maybe it is even cleaner to just use only use `self.trans_att`. This way we ensure the two never get out of sync.

And we catch all the keyword arguments to remain backwards compatible. And we could even raise a `PendingDepreciationWarning` untill version 2.0 is released.

**Comment by BSchilperoort** [comment]
Nice! `trans_att` would be a better (shorter!) name probably.

If the `transient_asym_att_x` and `matching_sections` kwargs become deprecated, it would also make sense to do the same for `sections`. It can be slightly confusing that you can define `ds.sections`, but also give the calibration the `sections` kwarg.

I think a `DeprecationWarning` is the usual error that's used though.

**Comment by bdestombe** [comment]
#128

**Comment by bdestombe** [comment]
Implemented in v2.0
